### PR TITLE
feat: auto generate/update locale files

### PR DESCRIPTION
**Pull request by ST-DDT**
This PR adds a script that automatically generates
- `src/locale/*.ts`
- `src/locales/index.ts`.
- and `docs/api/localization.md`

This should be merged after https://github.com/faker-js/faker/pull/248

The is the generation/update related part for #210

Currently this is not integrated in the build process.

### CODE REVIEW

File: scripts/generateLocales.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import fs from 'fs';
+import { format } from 'prettier';
+import options from '../.prettierrc.cjs';
+
+const pathLocale = 'src/locale/';
+const pathLocales = 'src/locales/';
```
Comment (Shinigami92) [inline scripts/generateLocales.ts:6]: You should use `resolve` for these

```ts
import { resolve } from 'node:path'

const pathLocale = resolve(__dirname, 'src', 'locale');
```

File: scripts/generateLocales.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import fs from 'fs';
```
Comment (Shinigami92) [inline scripts/generateLocales.ts:1]: We could use `import { readdirSync, writeFileSync } from 'node:fs';`

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

**Comment by ST-DDT** [comment]
This has been split into multiple PRs and needs to be updated once #263 and #264 have been merged.

**Comment by ST-DDT** [comment]
If the following change is considered breaking, then we could just temporarily remove/change it.
That way, we can still use the automation script.

````diff
- export = faker;
+ export default faker;
````

**Comment by Shinigami92** [comment]
I thing `export = faker` is currently the right way, so plain js node users can just use `const fakerDe = require('@faker-js/faker/locale/de')` instead of `const fakerDe = require('@faker-js/faker/locale/de').default`.
So indeed, for the `locale`s we have a special case here.

File: package.json [hunk 3]
```diff
@@ -55,6 +55,7 @@
     "build:code": "esno ./scripts/bundle.ts",
     "build:types": "tsc --emitDeclarationOnly --outDir dist/types",
     "build": "run-s build:clean build:code build:types",
+    "generate": "esno ./scripts/generateLocales.ts",
```
Comment (Shinigami92) [inline package.json:58]: Maybe we should be a bit more explizit here and name the script like `generate:locale`

File: src/locale/en.ts [hunk 4]
```diff
@@ -9,4 +9,4 @@ const faker = new Faker({
   },
 });
 
-export = faker;
```
Comment (Shinigami92) [inline src/locale/en.ts:12]: I think you missed that one, is it not generated?

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

**Review by Shinigami92 — DISMISSED** [review]
Nice work :rocket:

**Review by Shinigami92 — DISMISSED** [review]

File: scripts/generateLocales.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+let localeIndexLocales = 'const locales: KnownLocales = {\n';
+
+let localizationLocales = '| Locale | Name |\n| :--- | :--- |\n';
+
+const autoGeneratedCommentHeader = `/*
+ * This file is automatically generated.
+ * Run '${scriptCommand}' to update.
+ */`;
+
+for (let locale of locales) {
```
Comment (Shinigami92) [inline scripts/generateLocales.ts:34]:
```suggestion
for (const locale of locales) {
```

**Review by Shinigami92 — APPROVED** [review]

**Review by Shinigami92 — DISMISSED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Comment by ST-DDT** [comment]
I need this in main in order to update it for #363 .
(There is no automation enabled to this yet.)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -13,6 +13,7 @@ If you want to make `Faker` a better, please read the following contribution gui
   Make sure you **build** the project before running the docs, cause some files depend on `dist`.
   Use `pnpm run docs:dev` to edit them in live mode.
 - The tests are executing `vitest` against `test/**/*.spec.ts`
+- If you update the locales, make sure to run `pnpm run generate:locales` to generate/update the related files.
```
Comment (prisis) [inline CONTRIBUTING.md:16]: Can we create a ticket to add a workflow test for this? that is only running if something in the locales folder did change?

**Review by prisis — APPROVED** [review]

**Pull request merged**